src/runner/main.cpp

```diff
@@ -76,11 +62,61 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine
     Trace::EventLaunch(get_product_version());
 
     result = run_message_loop();
-  } catch (std::runtime_error& err) {
+  } catch (std::runtime_error & err) {
     std::string err_what = err.what();
-    MessageBoxW(NULL, std::wstring(err_what.begin(),err_what.end()).c_str(), L"Error", MB_OK | MB_ICONERROR);
+    MessageBoxW(NULL, std::wstring(err_what.begin(), err_what.end()).c_str(), L"Error", MB_OK | MB_ICONERROR);
     result = -1;
   }
   Trace::UnregisterProvider();
   return result;
 }
+
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow) {
+  WCHAR username[UNLEN + 1];
+  DWORD username_length = UNLEN + 1;
+  GetUserNameW(username, &username_length);
+  auto runner_mutex = CreateMutexW(NULL, TRUE, (std::wstring(L"Local\\PowerToyRunMutex") + username).c_str());
+  if (runner_mutex == NULL || GetLastError() == ERROR_ALREADY_EXISTS) {
+    // The app is already running
+    return 0;
+  }
+  int result = 0;
+  try {
+    // Singletons initialization order needs to be preserved, first events and
+    // then modules to guarantee the reverse destruction order.
+    SystemMenuHelperInstace();
+    powertoys_events();
+    modules();
+
+    auto general_settings = get_general_settings();
+    int rvalue = 0;
+    if (is_process_elevated() ||
+        general_settings.GetNamedBoolean(L"run_elevated") == false ||

```

Comment:
`general_settings.GetNamedBoolean(L"run_elevated", false)`?

Reply:
load_general_settings will set this, but I guess no harm in being protective here.